### PR TITLE
feat: Capture meta operator in support bundle

### PR DESCRIPTION
**Pull request by Elsie4ever**
---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to Azure IoT Operations tooling!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT Operations CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).

### CODE REVIEW

**Comment by Elsie4ever** [comment]
IT runs
image

File: azext_edge/edge/providers/support/meta.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        prefix_names=[META_PREFIX_NAMES],
+    )
+
+
+def fetch_pods(since_seconds: int = DAY_IN_SECONDS):
+    return process_v1_pods(
+        directory_path=META_DIRECTORY_PATH,
+        label_selector=META_NAME_LABEL,
+        since_seconds=since_seconds,
+        prefix_names=[META_PREFIX_NAMES],
```
Comment (digimaun) [inline azext_edge/edge/providers/support/meta.py:51]: Having this prefix name to filter on every resource type will be too limiting. For example, is the `aio-pre-install-job-*` captured in this PR ?

**Comment by Elsie4ever** [comment]
new IT runs
image

File: azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py [hunk 2]
```diff
@@ -113,8 +113,11 @@ def _get_expected_services(
         expected_services = OpsServiceType.list()
         expected_services.remove(OpsServiceType.auto.value)
         expected_services.append("otel")
-        # device registry folder will not be created if there are no device registry resources
```
Comment (vilit1) [inline azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py:116]: will the device registry folder be created even if there are no assets/asset endpoints now?

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py [hunk 3]
```diff
@@ -207,18 +207,22 @@ def get_file_map(
...
         ops_path = path.join(BASE_ZIP_PATH, namespace, ops_service)
         c_path = path.join(BASE_ZIP_PATH, c_namespace, "clusterconfig", ops_service)
         file_map["usage"] = convert_file_names(walk_result[c_path]["files"])
         file_map["__namespaces__"]["usage"] = c_namespace
-    elif ops_service != "otel":
+    elif ops_service == "deviceregistry":
         assert len(walk_result) == 1
+        if ops_path not in walk_result:
+            pytest.skip(f"No bundles created for {ops_service}.")
+    elif ops_service != "otel" and ops_service != "meta":
```
Comment (vilit1) [inline azext_edge/tests/edge/support/create_bundle_int/helpers.py:224]: is meta an --svc param or no?

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py [hunk 4]
```diff
@@ -207,18 +207,22 @@ def get_file_map(
...
     elif ops_service == "billing":
-        assert len(walk_result) == 2
+        assert len(walk_result) == 3
         ops_path = path.join(BASE_ZIP_PATH, namespace, ops_service)
         c_path = path.join(BASE_ZIP_PATH, c_namespace, "clusterconfig", ops_service)
         file_map["usage"] = convert_file_names(walk_result[c_path]["files"])
         file_map["__namespaces__"]["usage"] = c_namespace
-    elif ops_service != "otel":
+    elif ops_service == "deviceregistry":
         assert len(walk_result) == 1
```
Comment (vilit1) [inline azext_edge/tests/edge/support/create_bundle_int/helpers.py:221]: so if assets + endpoints are not created, this should be one (meta folder) but if they are created, this should be 2?

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py (hunk 3 above)
Comment (Elsie4ever) [reply azext_edge/tests/edge/support/create_bundle_int/helpers.py:224]: with current plan no

File: azext_edge/edge/providers/support/base.py [hunk 5]
```diff
@@ -480,3 +496,10 @@ def _process_kubernetes_resources(
         )
 
     return processed
+
+
+def exclude_resources_with_prefix(resources: List[dict], exclude_prefixes: List[str]) -> List[str]:
```
Comment (digimaun) [inline azext_edge/edge/providers/support/base.py:501]: This doesn't return List[str] does it. Also, resources, isn't being passed List[dict] right now.

Comment (Elsie4ever) [reply azext_edge/edge/providers/support/base.py:501]: yeah, thanks!

File: azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py (hunk 2 above)
Comment (Elsie4ever) [reply azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py:116]: the folder should only create when there is crd instance, let me double check

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py (hunk 4 above)
Comment (Elsie4ever) [reply azext_edge/tests/edge/support/create_bundle_int/helpers.py]: right, update and verify as we talked offline

File: azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py (hunk 2 above)
Comment (Elsie4ever) [reply azext_edge/tests/edge/support/create_bundle_int/test_auto_int.py:116]: checked it is created only when there is crd instance

File: azext_edge/edge/providers/support/meta.py (hunk 1 above)
Comment (Elsie4ever) [reply azext_edge/edge/providers/support/meta.py]: updated and verified the resources are being captured

**Comment by Elsie4ever** [comment]
with assetendpoint crd instance IT runs
image

File: azext_edge/tests/edge/support/create_bundle_int/helpers.py (hunk 4 above)
Comment (Elsie4ever) [reply azext_edge/tests/edge/support/create_bundle_int/helpers.py]: updated and verified the IT pass with/without DR crd

File: azext_edge/edge/providers/support/base.py [hunk 6]
```diff
@@ -62,6 +88,7 @@ def process_v1_pods(
     label_selector: Optional[str] = None,
     prefix_names: Optional[List[str]] = None,
     pod_prefix_for_init_container_logs: Optional[List[str]] = None,
+    exclude_prefixes: Optional[List[str]] = None,
 ) -> List[dict]:
     from kubernetes.client.models import V1Pod, V1PodList
```
Comment (digimaun) [inline azext_edge/edge/providers/support/base.py:93]: Should we also eliminate the redundant imports that are being handled at the higher scope? Like V1PodList here.

Comment (Elsie4ever) [reply azext_edge/edge/providers/support/base.py]: thanks, updated!

**Review by digimaun — APPROVED** [review]

**Pull request merged**